src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+
+Once published, you can see the publication event in the **Activity Log** on
+`pub.dev`.
+The **Activity Log** entry should contain a link to the GitHub Action run
+that published the package version.
+
+![Audit log after publishing from GitHub Actions](audit-log-pub-gh.png)
+
+## Hardening security of automated publishing
+
+### Hardening security with tag protection rules on GitHub
+
+
+Configuring automated publishing from GitHub Actions has one security
+flaw. Any GitHub user who can push a tag to your repository can publish
+your package.
+To restrict who can push tags to your repository, use
+[tag protection rules][5] on GitHub.
+
+By limiting who creates tags matching your `tag-pattern`,
+you limit who can publish the package.
+
+At this time, these [tag protection rules][5] lack flexibility. You may
+want to restrict who can trigger publishing using _GitHub Deployment
+environments_. The next section outlines this process.
+
+
+### Hardening security with GitHub Deployment Environments
+
+When configuring automated publishing from GitHub Actions in `pub.dev`,
+you can require a _[GitHub Actions environment][6]_.
+To use a GitHub Actions environment for publishing you must:
+
+1. Navigate to the **Admin** tab (`pub.dev/packages/<package>/admin`)
+   in `pub.dev`.
+1. Find the **Automated publishing** section.
+1. Click **Require GitHub Actions environment**.
+   The **Environment** field displays.
+
+    ![Configure pub.dev to require a GitHub deployment environment](pub-dev-gh-env-setup.png)
+
+1. Write an environment name into the **Environment** field.
+   To keep the process simple, use `pub.dev`.
+1. Click **Update**.
+
+When you configure an environment on `pub.dev`, you need to update the
+`publish.yml` file for GitHub Action workflow in your repository.
+You need to add the `environment` key with a value that matches the
+**Environment** field value in `pub.dev`.
+
+1. [Create an environment][8] in your GitHub repository.
+   This environment must use the same name you provided in `pub.dev`.
+1. Open your `.github/workflows/publish.yml` workflow file in your
+   preferred editor.
+1. Add the `environment` key with the corresponding value.
+   The result should resemble the following file:
+
+    ```yaml
+    # .github/workflows/publish.yml
+    name: Publish to pub.dev
+
+    on:
+      push:
+        tags:
+        - 'v[0-9]+.[0-9]+.[0-9]+*' # for tags like: 'v1.2.3'
+
+    jobs:
+      publish:
+        uses: dart-lang/setup-dart/.github/workflows/publish.yml
+        with:
+          # Specify the github actions deployment environment
+          environment: pub.dev
+          # working-directory: path/to/package/within/repository
+    ```
+
+The temporary [GitHub-signed OpenID Connect token][1] includes the
+environment value used to authenticate with `pub.dev`. 
+This prevents a user with push permission on your repository from
+modifying the workflow file to circumvent
+[environment protection rules][7].
```

Comment:
No it doesn't!

You user can still modify the workflow file, but modifying the file doesn't circumvent environment protection rules.